pkg/providers/cloudstack/cloudstack.go

```diff
@@ -87,9 +91,56 @@ func (p *cloudstackProvider) PostWorkloadInit(ctx context.Context, cluster *type
 }
 
 func (p *cloudstackProvider) UpdateSecrets(ctx context.Context, cluster *types.Cluster) error {
+	contents, err := p.createSecrets(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("creating secrets object: %v", err)
+	}
+
+	err = p.providerKubectlClient.ApplyKubeSpecFromBytes(ctx, cluster, contents)
+	if err != nil {
+		return fmt.Errorf("applying secrets object: %v", err)
+	}
 	return nil
 }
 
+func (p *cloudstackProvider) createSecrets(ctx context.Context, cluster *types.Cluster) ([]byte, error) {
+	if err := p.providerKubectlClient.CreateNamespaceIfNotExists(ctx, cluster.KubeconfigFile, constants.EksaSystemNamespace); err != nil {
+		return nil, err
+	}
+
+	secrets := [][]byte{}
+	for _, profile := range p.execConfig.Profiles {
+		secret := createSecret(profile)
+		if bytes, err := yaml.Marshal(secret); err == nil {
+			secrets = append(secrets, bytes)
+		} else {
+			return nil, fmt.Errorf("marshalling secret: %v: %v", secret, err)
```

Comment:
I'm not sure if we want to log the secret, maybe just the error

Reply:
Fixed